contributors/devel/sig-architecture/api-conventions.md

```diff
@@ -742,51 +768,176 @@ have a built-in `nil` value.
 
 ## Defaulting
 
-Default resource values are API version-specific, and they are applied during
-the conversion from API-versioned declarative configuration to internal objects
-representing the desired state (`Spec`) of the resource. Subsequent GETs of the
-resource will include the default values explicitly.
-
-Incorporating the default values into the `Spec` ensures that `Spec` depicts the
-full desired state so that it is easier for the system to determine how to
-achieve the state, and for the user to know what to anticipate.
+In general we want default values to be explicitly represented in our APIs,
+rather than asserting that "unspecified fields get the default behavior".  This
+is important so that:
+ - default values can evolve and change in newer API versions
+ - the stored configuration depicts the full desired state, making it easier
+   for the system to determine how to achieve the state, and for the user to
+   know what to anticipate
+
+There are 3 distinct ways that default values can be applied when creating or
+updating (including patch and apply) a resource:
+
+ 1. static: based on the requested API version and possibly other fields in the
+    resource, fields can be assigned values during the API call
+ 2. admission control: based on the configured admission controllers and
+    possibly other state in or out of the cluster, fields can be assigned
+    values during the API call
+ 3. controllers: arbitrary changes (within the bounds of what is allowed) can
+    be made to a resource after the API call has completed
+
+Some care is required when deciding which mechanism to use and managing the
+semantics.
+
+### Static Defaults
```

Comment:
Would it be worth describing what CEL will now allow? @jpbetz And it's probably a little premature but also possibly in the admission webhook category below.

Reply:
Good question.  CEL for admission control (only an idea right now, not even a provisional KEP) would be the best fit and would align with the below "Admission Controlled Defaults".  I would just be another mechanism, in addition to admission controllers and admission webhooks to perform the mutation.

Reply:
can we take that as a followup (since I myself and not sure what to write..)?

Reply:
I'm totally fine with that, thanks!